src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -34394,6 +34395,40 @@ static void G(string? s)
             F(t);
             F(s);
         }
+        else
+        {
+            F(s); // warn
+        }
+    }
+}";
+            var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (13,15): warning CS8604: Possible null reference argument for parameter 's' in 'void C.F(string s)'.
+                //             F(s); // warn
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "s").WithArguments("s", "void C.F(string s)").WithLocation(13, 15)
+                );
+        }
+
+        [Fact]
+        [WorkItem(30310, "https://github.com/dotnet/roslyn/issues/30310")]
+        public void IsPattern_02_WithNullTest()
+        {
+            var source =
+@"class C
+{
+    static void F(string s) { }
+    static void G(string? s)
+    {
+        if (s == null) throw null;
+        if (s is string t)
+        {
+            F(t);
+            F(s);
+        }
+        else
+        {
+            F(s);
+        }
```

Comment:
Is this distinct from the `F4` tests in `IsDeclarationPattern_01`, `_02`, `_03`?

Reply:
Hum, it's very slightly different, but not significantly. Thanks
With this I think we already have sufficient coverage. I'll just close the PR and the bug.